### PR TITLE
Add SendGridClient

### PR DESCRIPTION
**Pull request by elisabeth-shakespeare**
add SendGridClient with function SendEmail

### CODE REVIEW

File: src/Aquifer.Common/Clients/SendGridClient.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+using Aquifer.Data;
+using Microsoft.Extensions.Configuration;
+using SendGrid;
+using SendGrid.Helpers.Mail;
+
+namespace Aquifer.Common.Clients;
+
+public interface ISendGridClient
+{
+    Task<Response> SendEmail(AquiferDbContext dbContext, SendGridEmailConfiguration emailConfiguration, CancellationToken ct);
```
Comment (jwinston-bn) [inline src/Aquifer.Common/Clients/SendGridClient.cs:10]: I'd inject the AquiferDbContext in the constructor.

Comment (jwinston-bn) [reply src/Aquifer.Common/Clients/SendGridClient.cs:10]: though I'm not even seeing where it's being used

Comment (elisabeth-shakespeare) [reply src/Aquifer.Common/Clients/SendGridClient.cs:10]: Thanks for catching that. I don't use it. I'll take that out.

File: src/Aquifer.Common/Clients/SendGridClient.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    private readonly string _apiToken;
+
+    public SendGridClient(IAzureKeyVaultClient keyVaultClient, IConfiguration configuration)
+    {
+        _apiToken = keyVaultClient.GetSecretAsync(ApiKeySecretName).GetAwaiter().GetResult();
+    }
+
+    public async Task<Response> SendEmail(AquiferDbContext dbContext, SendGridEmailConfiguration emailConfiguration, CancellationToken ct)
+    {
+        var client = new SendGrid.SendGridClient(_apiToken);
```
Comment (jwinston-bn) [inline src/Aquifer.Common/Clients/SendGridClient.cs:35]: The whole point of making this a singleton is to new this up once.

Comment (elisabeth-shakespeare) [reply src/Aquifer.Common/Clients/SendGridClient.cs:35]: This is from the SendGrid library, not the Client that I've written.

File: src/Aquifer.Common/Clients/SendGridClient.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    public required string Subject { get; set; }
+    public required string HtmlContent { get; set; }
+    public string? PlainTextContent { get; set; } = null!;
+    public required List<EmailAddress> ToAddresses { get; set; }
+}
+
+public class SendGridClient : ISendGridClient
+{
+    private const string ApiKeySecretName = "SendGridMarketingApiKey";
+    private readonly string _apiToken;
```
Comment (jwinston-bn) [inline src/Aquifer.Common/Clients/SendGridClient.cs:26]: This can be local to the constructor

**Review by jwinston-bn — APPROVED** [review]

**Pull request merged**